mindmaps-test/src/test/java/io/mindmaps/graql/internal/analytics/AnalyticsTest.java

```diff
@@ -858,14 +858,383 @@ public void testDegreeIsCorrectAssertionAboutAssertion() throws MindmapsValidati
+        EntityType animal = transaction.putEntityType("animal").playsRole(pet);
+
+        // make one person breeder and owner
+        Entity coco = transaction.putEntity("coco", animal);
+        Entity dave = transaction.putEntity("dave", person);
+        Map<RoleType,Instance> roleMap = new HashMap<>();
+        roleMap.put(pet,coco);
+        roleMap.put(owner,dave);
+
+        Relation daveBreedsAndOwnsCoco = transaction.putRelation(mansBestFriend,roleMap);
+
+        // manual degrees
+        Map<String,Long> referenceDegrees = new HashMap<>();
+        referenceDegrees.put(coco.getId(),1L);
+        referenceDegrees.put(dave.getId(),1L);
+        referenceDegrees.put(daveBreedsAndOwnsCoco.getId(),2L);
+
+        // create resource to persist on
+        RoleType value = transaction.putRoleType("value");
+        RoleType target = transaction.putRoleType("target");
+        RelationType hasResource = transaction.putRelationType("has-resource").hasRole(target).hasRole(value);
+        ResourceType<Long> degreeResource = transaction.putResourceType("degree-resource",Data.LONG).playsRole(value);
+
+        Set<Type> ct = new HashSet<>();
+
+        // validate
+        transaction.commit();
+
+        mansBestFriend = transaction.getRelationType("mans-best-friend");
+        person = transaction.getEntityType("person");
+        animal = transaction.getEntityType("animal");
+        ct.clear();
+        ct.add(mansBestFriend);
+        ct.add(person);
+        ct.add(animal);
+
+        // compute and persist degrees
+        Analytics analytics = new Analytics(ct);
+        value = transaction.getRoleType("value");
+        degreeResource = transaction.getResourceType("degree-resource");
+//        analytics.degreesAndPersist(value, degreeResource);
+
+        // check degrees are correct
+        referenceDegrees.entrySet().forEach(entry->{
+            Instance instance = transaction.getInstance(entry.getKey());
+            if (instance.isEntity()) {
+                assertTrue(instance.asEntity().resources().iterator().next().getValue().equals(entry.getValue()));
+            } else if (instance.isRelation()) {
+                assertTrue(instance.asRelation().resources().iterator().next().getValue().equals(entry.getValue()));
+            }
+        });
+
+        // check only expected resources exist
+        Collection<String> allConcepts = new ArrayList<>();
+        ResourceType<Long> rt = transaction.getResourceType("degree-resource");
+        Collection<Resource<Long>> degrees = rt.instances();
+        degrees.forEach(i->i.ownerInstances().iterator().forEachRemaining(r ->
+                allConcepts.add(r.getId())));
+
+        Map<String,Long> allResources = new HashMap<>();
+//        Set<String> nodesWithDegree = sg.getInstanceIds();
+        Collection<Resource<?>> resources=null;
+//        for (String id : nodesWithDegree) {
```

Comment:
Get rid of giant comment blocks if you can. Unless they are supremely necessary.